lms/static/scripts/frontend_apps/components/StatusChip.tsx

```diff
@@ -0,0 +1,35 @@
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+
+export type StatusChipProps = {
+  /**
+   * Determines the aspect of the status chip, being "success" the one
+   * representing the most successful status, and "fatal" the one representing
+   * a total failure.
+   */
+  variant: 'success' | 'semi-success' | 'warning' | 'error' | 'fatal';
+
+  children?: ComponentChildren;
+};
+
+/**
+ * A status chip (AKA label or tag) that visually represents the success level
+ * on a scale of 1 to 5
+ */
+export default function StatusChip({ variant, children }: StatusChipProps) {
+  return (
+    <div
+      className={classnames('rounded font-bold inline-block px-2 py-0.5', {
+        'bg-green-success text-white': variant === 'success',
+        // 'bg-green-700 text-white': variant === 'success',
+        'bg-green-200 text-green-900': variant === 'semi-success',
+        'bg-amber-100 text-amber-900': variant === 'warning',
+        'bg-red-200 text-red-900': variant === 'error',
+        'bg-red-error text-white': variant === 'fatal',
+        // 'bg-red-600 text-white': variant === 'fatal',
```

Comment:
The commented-put styles pass axe-core accessibility checks on contrast, but do not follow our standard color palette and designs.